api/src/pcapi/scripts/add_venue_providers_on_venue_with_siret/main.py

```diff
@@ -0,0 +1,344 @@
+    )
+
+
+@atomic()
+def main(not_dry: bool) -> None:
+    log_modifications: list[tuple[int, str]] = []
+    log_fails: list[tuple[int, str]] = []
+    venues_without_siret_with_providers = _get_venues_without_siret_and_not_permanent_with_providers()
+    venues_with_siret_by_offerer = _get_unique_venues_with_siret_by_offerer()
+    user = User.query.get(MIGRATION_AUTHOR_ID)
+    venue_with_siret = {venue.managingOffererId: venue for venue in venues_with_siret_by_offerer}
+    for venue_source in venues_without_siret_with_providers:
+        venue_destination = venue_with_siret.get(venue_source.managingOffererId)
+        if not venue_destination:
+            log_fails.append((venue_source.id, f"Pas de siret sur cet offerer {venue_source.managingOffererId}"))
+            continue
+
+        for venue_provider in venue_source.venueProviders:
+            try:
+                with atomic():
+                    _create_venue_provider(venue_source, venue_destination, venue_provider, user)
+                    log_modifications.append(
+                        (
+                            venue_source,
+                            f"Ajout du provider {venue_provider.provider} sur la venue {venue_destination}",
+                        )
+                    )
+                    logger.info(
+                        "Adding provider %s on venue %s",
+                        venue_provider.providerId,
+                        venue_destination.id,
+                    )
+            except sa_exc.IntegrityError:
+                logger.error("IntegrityError: %s is already synch. with %s", venue_destination, venue_provider.provider)
+                log_fails.append(
+                    (venue_source.id, f"Cette venue est déjà synchro avec le provider {venue_provider.provider}.")
+                )
+                continue
+            except ProviderException:
+                logger.error(
+                    "Provider exception: could not move provider %s to venue %s",
+                    venue_provider.provider,
+                    venue_destination,
+                )
+                log_fails.append(
+                    (
+                        venue_source.id,
+                        f"Impossible de configurer le Provider {venue_provider.provider} avec cette venue",
+                    )
+                )
+                continue
+
+    _write_modifications(log_modifications, "add_providers.csv")
+    _write_modifications(log_fails, "add_providers_fails.csv")
+
+    if not not_dry:
+        mark_transaction_as_invalid()
+
+
+if __name__ == "__main__":
+    app.app_context().push()
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--not-dry", action="store_true")
+    args = parser.parse_args()
+
+    main(not_dry=args.not_dry)
+
+    if args.not_dry:
+        logger.info("Finished")
+        db.session.commit()
+    else:
+        logger.info("Finished dry run, rollback")
+        db.session.rollback()
```

Comment:
Il me semble que tu peux supprimer ce block vu que tu appliques `@atomic()` à `main`
```suggestion
    if args.not_dry:
        logger.info("Finished")
        db.session.commit()
    else:
        logger.info("Finished dry run, rollback")
        db.session.rollback()
```